cpp/src/parquet/size_statistics.cc

```diff
@@ -64,14 +64,19 @@ void SizeStatistics::IncrementUnencodedByteArrayDataBytes(int64_t value) {
 }
 
 void SizeStatistics::Validate(const ColumnDescriptor* descr) const {
-  if (repetition_level_histogram.size() !=
-      static_cast<size_t>(descr->max_repetition_level() + 1)) {
-    throw ParquetException("Repetition level histogram size mismatch");
-  }
-  if (definition_level_histogram.size() !=
-      static_cast<size_t>(descr->max_definition_level() + 1)) {
-    throw ParquetException("Definition level histogram size mismatch");
-  }
+  auto validate_histogram = [](const std::vector<int64_t>& histogram, int16_t max_level,
+                               const std::string& name) {
+    if (max_level == 0 && histogram.empty()) {
+      return;
+    }
+    if (histogram.size() != static_cast<size_t>(max_level + 1)) {
+      throw ParquetException(name + " level histogram size mismatch");
+    }
+  };
+  validate_histogram(repetition_level_histogram, descr->max_repetition_level(),
+                     "Repetition");
+  validate_histogram(definition_level_histogram, descr->max_definition_level(),
+                     "Definition");
```

Comment:
You may want to add a threshold argument to `validate_histogram`, the reason being that the definition level histogram can be omitted for max def equal 0 _or_ 1. arrow-rs and cudf both use the same cutoff of `0` on write (not sure what parquet-java does, but you wrote that 😄), but down the road someone might stick to the spec and this test will still fail.